nuxeo-features/rest-api/nuxeo-rest-api-test/src/test/java/org/nuxeo/ecm/restapi/test/enrichers/FailingEnricherTest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * (C) Copyright 2020 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Nour "AIR" AL KOTOB
+ */
+
+package org.nuxeo.ecm.restapi.test.enrichers;
+
+import static org.junit.Assert.fail;
+
+import javax.inject.Inject;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.PathRef;
+import org.nuxeo.ecm.core.io.marshallers.json.AbstractJsonWriterTest;
+import org.nuxeo.ecm.core.io.marshallers.json.document.DocumentModelJsonWriter;
+import org.nuxeo.ecm.core.io.registry.context.RenderingContext.CtxBuilder;
+import org.nuxeo.ecm.core.test.annotations.Granularity;
+import org.nuxeo.ecm.core.test.annotations.RepositoryConfig;
+import org.nuxeo.ecm.restapi.test.RestServerFeature;
+import org.nuxeo.ecm.restapi.test.RestServerInit;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+
+@RunWith(FeaturesRunner.class)
+@Features({ RestServerFeature.class })
+@RepositoryConfig(cleanup = Granularity.METHOD, init = RestServerInit.class)
+@Deploy("org.nuxeo.ecm.platform.restapi.test.test:enrichers-contrib.xml")
+public class FailingEnricherTest
+        extends AbstractJsonWriterTest.External<DocumentModelJsonWriter, DocumentModel> {
+
+    public FailingEnricherTest() {
+        super(DocumentModelJsonWriter.class, DocumentModel.class);
+    }
+
+    @Inject
+    private CoreSession session;
+
+    @Test
+    public void test() {
+        DocumentModel root = session.getDocument(new PathRef("/"));
+        // Use the given URL codec name
+        try {
+            jsonAssert(root, CtxBuilder.enrichDoc("failingEnricher").get());
+        } catch (Exception e) {
```

Comment:
No need for try/catch, the test itself failing is enough (and more informative with respect to the exception).

Reply:
👍 I also added a logCatcher as it is requested explicitly to log the error.